File: src/main/java/cn/ruanyun/backInterface/modules/business/kotlinDemo/controller/KotlinDemoController.java
package cn.ruanyun.backInterface.modules.business.kotlinDemo.controller;

import cn.ruanyun.backInterface.common.utils.ResultUtil;
import cn.ruanyun.backInterface.common.vo.Result;
import cn.ruanyun.backInterface.modules.business.kotlinDemo.pojo.KotlinDemo;
import cn.ruanyun.backInterface.modules.business.kotlinDemo.service.IKotlinDemoService;
import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

/**
 * @author z
 * kotlin栗子管理接口
 */
@Slf4j
@RestController
@RequestMapping("/ruanyun/kotlinDemo")
@Transactional
public class KotlinDemoController {

    @Autowired
    private IKotlinDemoService iKotlinDemoService;


   /**
     * 更新或者插入数据
     * @param kotlinDemo
     * @return
    */
    @PostMapping(value = "/insertOrderUpdateKotlinDemo")
    public Result<Object> insertOrderUpdateKotlinDemo(KotlinDemo kotlinDemo){

        try {

            iKotlinDemoService.insertOrderUpdateKotlinDemo(kotlinDemo);
            return new ResultUtil<>().setSuccessMsg("插入或者更新成功!");
        }catch (Exception e) {

            return new ResultUtil<>().setErrorMsg(201, e.getMessage());
        }
    }


    /**
     * 移除数据
     * @param ids
     * @return
    */
    @PostMapping(value = "/removeKotlinDemo")
    public Result<Object> removeKotlinDemo(String ids){

        try {

            iKotlinDemoService.removeKotlinDemo(ids);
            return new ResultUtil<>().setSuccessMsg("移除成功！");
        }catch (Exception e) {

            return new ResultUtil<>().setErrorMsg(201, e.getMessage());
        }
    }

}

File: src/main/java/cn/ruanyun/backInterface/modules/business/kotlinDemo/controller/KotlinDemoController.kt
package cn.ruanyun.backInterface.modules.business.kotlinDemo.controller

import cn.ruanyun.backInterface.common.utils.ResultUtil
import cn.ruanyun.backInterface.common.vo.Result
import cn.ruanyun.backInterface.modules.business.kotlinDemo.pojo.KotlinDemo
import cn.ruanyun.backInterface.modules.business.kotlinDemo.service.KotlinDemoService
import io.swagger.annotations.Api
import io.swagger.annotations.ApiOperation
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/kotlin")
@Api(tags = arrayOf("测试kotlin"))
class KotlinDemoController {

    @Autowired
    lateinit var kotlinDemoService: KotlinDemoService

    @PostMapping("/saveKotlin")
    @ApiOperation("插入kotlin")
    fun saveKotlin(kotlinDemo: KotlinDemo) : Result<Any> {

        kotlinDemoService.saveKotlin(kotlinDemo)

        return ResultUtil<Any>().setSuccessMsg("插入测试成功！")
    }

}